src/ApplicationInsights.Kubernetes/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -1,29 +1,98 @@
 ﻿using System;
-using System.Threading.Tasks;
+using System.Linq;
 using Microsoft.ApplicationInsights.Extensibility;
 using Microsoft.ApplicationInsights.Kubernetes;
+using Microsoft.ApplicationInsights.Kubernetes.Utilities;
+using Microsoft.Extensions.Logging;
+
+using static Microsoft.ApplicationInsights.Kubernetes.StringUtils;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
-
     /// <summary>
     /// Extnesion method to inject Kubernetes Telemtry Initializer.
     /// </summary>
     public static class ApplicationInsightsExtensions
     {
+        /// <summary>
+        /// Enable Application Insights Kubernetes for the Default TelemtryConfiguration in the dependency injection system.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="timeout"></param>
+        /// <param name="kubernetesServiceCollectionBuilder"></param>
+        /// <returns></returns>
         public static IServiceCollection EnableKubernetes(
-            this IServiceCollection services, 
+            this IServiceCollection services,
+            TimeSpan? timeout = null,
+            IKubernetesServiceCollectionBuilder kubernetesServiceCollectionBuilder = null)
+        {
+            // Inject of the service shall return immediately.
+            EnableKubernetesImpl(services, timeout, kubernetesServiceCollectionBuilder);
+            return services;
+        }
+
+        /// <summary>
+        /// Enable Application Insights Kubernetes for a given TelemetryConfiguration.
+        /// Note: The use of EnableKubernetes() on the ServiceCollection is always preferred unless you have more than one
+        /// TelemetryConfiguration instances.
+        /// </summary>
+        public static void EnableKubernetes(
+            this TelemetryConfiguration telemetryConfiguration,
+            TimeSpan? timeout = null,
+            IKubernetesServiceCollectionBuilder kubernetesServiceCollectionBuilder = null)
+        {
+            if (telemetryConfiguration == null)
+            {
+                return;
+            }
+
+            IServiceCollection standaloneServiceCollection = new ServiceCollection();
+            standaloneServiceCollection = EnableKubernetesImpl(standaloneServiceCollection, timeout, kubernetesServiceCollectionBuilder);
+
+            IServiceProvider serviceProvider = standaloneServiceCollection.BuildServiceProvider();
+
+            ITelemetryInitializer k8sTelemetryInitializer = serviceProvider.GetServices<ITelemetryInitializer>()
+                .First(ti => ti.GetType() == typeof(KubernetesTelemetryInitializer));
```

Comment:
I assume injecting K8s telemetry initializer may fail e.g. due to timeout. In that case there will be no telemetry initializer and `First()` will throw.

Suggestion: use `FirstOrDefault()` and check for null

Reply:
1. I thought First<T>() makes sense as well. Unfortunately, First<T>() is an extension method of IEnumerable<T>. Omitting T only because of the Type Inference.
2. I think you are saying the constructor of KubernetesTelemetryInitializer() is blocking and will wait until the Kubernetes info to become available, thus, could timeout, right? That is not the case. The small design here is: The constructor always construct the object and return without being blocked:
```csharp
        public KubernetesTelemetryInitializer(
            IK8sEnvironmentFactory k8sEnvFactory,
            TimeSpan timeout,
            SDKVersionUtils sdkVersionUtils,
            ILogger<KubernetesTelemetryInitializer> logger)
        {
            _k8sEnvironment = null;
            _logger = logger;
            _sdkVersionUtils = Arguments.IsNotNull(sdkVersionUtils, nameof(sdkVersionUtils));
            _timeoutAt = DateTime.Now.Add(Arguments.IsNotNull(timeout, nameof(timeout)));
            _k8sEnvFactory = Arguments.IsNotNull(k8sEnvFactory, nameof(k8sEnvFactory));

            var _forget = SetK8sEnvironment();
        }
```
All these are synchronize operations and will finish the construction of the initializer _immediately_.
With all that said, as a public method, it is no guarantee that the required parameters won't be null. I think it a good idea to use FirstOrDefault() and check for null.
Will fix in i2.